crates/wasi-nn/src/backend/kserve.rs

```diff
@@ -0,0 +1,1012 @@
+                        (n.as_u64().unwrap()).to_le_bytes().as_slice(),
+                    ),
+                    KServeDatatype::INT8 => copy_to_destination(
+                        destination,
+                        &mut start_index,
+                        (n.as_i64().unwrap() as i8).to_le_bytes().as_slice(),
+                    ),
+                    KServeDatatype::INT16 => copy_to_destination(
+                        destination,
+                        &mut start_index,
+                        (n.as_i64().unwrap() as i16).to_le_bytes().as_slice(),
+                    ),
+                    KServeDatatype::INT32 => copy_to_destination(
+                        destination,
+                        &mut start_index,
+                        (n.as_i64().unwrap() as i32).to_le_bytes().as_slice(),
+                    ),
+                    KServeDatatype::INT64 => copy_to_destination(
+                        destination,
+                        &mut start_index,
+                        (n.as_i64().unwrap() as i64).to_le_bytes().as_slice(),
+                    ),
+                    KServeDatatype::FP32 => copy_to_destination(
+                        destination,
+                        &mut start_index,
+                        (n.as_f64().unwrap() as f32).to_le_bytes().as_slice(),
+                    ),
+                    KServeDatatype::FP64 => copy_to_destination(
+                        destination,
+                        &mut start_index,
+                        (n.as_f64().unwrap()).to_le_bytes().as_slice(),
+                    ),
+                    _ => panic!("Unsupported kserve datatype for rust"),
+                },
+                KServeTensorElement::String(s) => {
+                    copy_to_destination(destination, &mut start_index, s.as_bytes())
+                }
+            };
+        }
+        Ok(total_size as u32)
+    }
+}
+
+#[async_trait]
+impl GraphRegistry for KServeBackend {
+    async fn get_mut(&mut self, name: &str) -> Result<Option<&mut Graph>, BackendError> {
+        let graph_present = self.registry.contains_key(name);
+        let model_name = name.to_string();
+
+        //We don't already have the graph retrieve the model metadata to make sure it exists
+        if !graph_present {
+            let mut client = KServeClient::new(&self.server_url).await;
+            let model_metadata = client.get_model_metadata(&name.to_string()).await?;
+
+            let g = Arc::new(KServeGraph {
+                model_name: name.to_string(),
+                server_url: self.server_url.clone(),
+            });
+            let graph: Graph = Graph(g);
+            self.registry.insert(model_metadata.name.clone(), graph);
+        }
+
+        Ok(self.registry.get_mut(&model_name))
+
+        //This has to be here, because you can't do a mutable borrow twice.
+    }
+}
+
+fn copy_to_destination(destination: &mut [u8], start_index: &mut usize, src: &[u8]) {
```

Comment:
Instead of this helper function, use a `std::io::Cursor` and then the `byteorder::WriteBytesExt` trait at each of the call sites.